frontend/components/Course/Analytics/Cards/SummaryCards.tsx

```diff
@@ -0,0 +1,128 @@
+import { Card, Dropdown, Header, Segment } from "semantic-ui-react";
+import React, { useState } from "react";
+import AnalyticsCard from "./AnalyticsCard";
+import { useStatistic } from "../../../../hooks/data-fetching/analytics";
+import { Metric } from "../../../../types";
+
+interface SummaryCardsProps {
+    courseId: number;
+    queueId: number;
+}
+
+export default function SummaryCards({ courseId, queueId }: SummaryCardsProps) {
+    const [timeRange, setTimeRange] = useState(7);
+    const timeRangeOptions = [7, 30, 90].map((range) => {
+        return {
+            key: range,
+            value: range,
+            text: `Last ${range} days`,
+        };
+    });
+
+    const {
+        data: numAnsweredData,
+        isValidating: numAnsweredValidating,
+    } = useStatistic(courseId, queueId, Metric.NUM_ANSWERED, timeRange);
+
+    const { data: avgWaitData, isValidating: avgWaitValidating } = useStatistic(
+        courseId,
+        queueId,
+        Metric.AVG_WAIT,
+        timeRange
+    );
+
+    const {
+        data: studentsHelpedData,
+        isValidating: studentsHelpedValidating,
+    } = useStatistic(courseId, queueId, Metric.STUDENTS_HELPED, timeRange);
+
+    const {
+        data: avgTimeHelpingData,
+        isValidating: avgTimeHelpingValidating,
+    } = useStatistic(courseId, queueId, Metric.AVG_TIME_HELPING, timeRange);
+
+    const sum = (data) => {
+        return data.reduce((prev, cur) => prev + parseInt(cur.value, 10), 0);
+    };
+
+    const averageWeighted = (data, weights) => {
+        const timeHelping: { value: number; date: string }[] = [];
+        const numHelped: { value: number; date: string }[] = [];
```

Comment:
~~Honestly not much of an efficiency boost with only 90 items but it might be more readable to~~
~~- put data into a map, with key = dataItem.date and value = dataItem.value~~
~~- put weights into a map also with key = weightItem.date and value = weightItem.value~~
~~- loop through entryset and calculate sums~~

~~Actually... you don't have to rewrite the whole thing unless your CIS 121 OCD is bothering you...~~

Sorry ignore the above, overhead definitely not worth it

You can just get rid of the timeHelping and numHelped arrays and keep instead a dataSum += dataItem.value * weightItem.value and  weightSum += weightItem.value right?

Haha hopefully I didn't horribly misread the code

Reply:
yep you're right. definitely overcomplicated this.